[AC-10: Feature] Refactor register feature

turnkevin opened the pull request:
### Provide description as to what you have done in this pr.

Create signup page and do the localisation. 

### Provide screenshots or videos that the function is working.
![image](https://user-images.githubusercontent.com/21169466/171620668-29ec8ebf-7493-412a-84c0-5270afafcb8f.png)
### Provide screenshots of no warning, error in console.
![image](https://user-images.githubusercontent.com/21169466/171621210-832ebeb5-f0e1-4c79-a0e8-0516be21540b.png)

### Developer Checklist

#### When you finished, you will want to make sure you have...

- [x] No merge conflict with `develop` branch.
- [x] Check PR title is correct, (i.e `[AC-xx: Feature] Create pull request template`).
- [x] Check your app will run without crashes.
- [x] Check if there's hard coded values.
- [x] No `//eslint-disable-next-line no-unused-vars`, unless otherwise specified.
- [x] No unnecessary spaces between lines of code.
- [x] Check you are using `camelCase && PascalCase` correctly.
- [x] Check you named your code properly.
- [x] Remove unused comments.

turnkevin committed:
signup page

turnkevin committed:
login

turnkevin committed:
kevin

turnkevin committed:
add login main page

turnkevin committed:
create signup form, add react useform hook

turnkevin committed:
add yup and hook form

turnkevin committed:
add yup and hook form

turnkevin committed:
yarn update

turnkevin committed:
Change component name

turnkevin committed:
Change component name

turnkevin committed:
change component name

turnkevin committed:
removed merge conflicts

turnkevin committed:
Rename Navlogo to NavLogo

turnkevin committed:
Merge branch 'develop' into feature/AC-10-create-user-signup-login-flow

turnkevin committed:
Delete unused files

turnkevin committed:
Add signup page into app

turnkevin committed:
Refactor sign up form

turnkevin committed:
Rename SignupForm to SignUpForm

turnkevin committed:
rename SignupForm to SignUpForm

turnkevin committed:
Update translation for sign up page and nav bar component

turnkevin committed:
Add policy component to sign up page

turnkevin committed:
Connect signup page to API

turnkevin committed:
Delete unused img

turnkevin committed:
Change env

turnkevin committed:
Merge branch 'develop' into feature/AC-10-create-user-signup-login-flow

turnkevin committed:
Refactor signup page with accept policy

turnkevin committed:
Delete unused files

NorrisWu0 commented on .vscode/settings.json:
```diff
@@ -0,0 +1,3 @@
+{
+  "jira-plugin.workingProject": ""
```
What is this?

turnkevin replied:
I have already cancelled it.

NorrisWu0 commented on .env.development:
```diff
@@ -0,0 +1,4 @@
+NEXT_PUBLIC_GRAPHQL_ENDPOINT=http://a-comosus.com:3100/graphql/
```
I don't think we need to commit this, but it can stay here for now.

NorrisWu0 commented on public/locales/cn/auth.json:
```diff
@@ -21,5 +21,41 @@
+        "required": "你不要密码吗？"
+      }
+    },
+    "error": "要嘛不是用户名不对，就是密码不对。",
```
I don't think you need this error here.

NorrisWu0 commented on public/locales/cn/auth.json:
```diff
@@ -21,5 +21,41 @@
+      }
+    },
+    "error": "要嘛不是用户名不对，就是密码不对。",
+    "user-exist": "用户已存在，请换用户名。",
```
Imagine our backend will return different errors. We can group this type of message under `error` object.

```
{
  error: {
    existing-user.
    ...other rule,
  }
}

NorrisWu0 commented on public/locales/cn/auth.json:
```diff
@@ -21,5 +21,41 @@
+      "action": "创建一个吧!"
+    },
+    "policy": {
+      "title": "创建帐户即表示您同意我们的条款和条件。查看我们的隐私声明，了解我们如何管理您的个人信息。",
```
u sure this is `title`?

NorrisWu0 commented on src/common/components/generics/Policy.tsx:
```diff
@@ -0,0 +1,23 @@
+import { Controller } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+
+export default function Policy({ name, control }) {
```
So there are two different scenarios here.

if we are reusing this component else where, should it really be called `Policy`?

If we are not reusing it, then should it be a component under `@common` or does it belongs to its respective module?

NorrisWu0 commented on src/common/components/index.ts:
```diff
@@ -7,7 +7,7 @@ export { default as NavBar } from './generics/NavBar';
 export { default as SelectLanguage } from './generics/SelectLanguage';
 export { default as ToggleThemeButton } from './generics/ToggleThemeButton';
-
+export { default as Policy } from './generics/Policy';
```
Same as above

NorrisWu0 commented on src/modules/auth/LoginForm/login.gql:
```diff
@@ -7,3 +7,9 @@ mutation Login($payload: LoginDetailInput!) {
   }
 }
+
+mutation Register($payload: RegisterDetailInput!) {
```
You can either rename this file to `auth.gql` or create a separate `signup.gql`

NorrisWu0 commented on src/styles/Home.module.css:
```diff
@@ -0,0 +1 @@
+
```
?

turnkevin committed:
Delete unused file

turnkevin committed:
Refactor localisation

turnkevin committed:
Relocate signup policy

turnkevin committed:
Relocate signup mutation

NorrisWu0 committed:
Fix `<Checkbox>` props type error

Checkbox props by default only accept type of `string | number` for field `value`. Therefore, `SignUpFormTypes` is updated to match the required type. And transforming the value to `boolean` when making mutation.

NorrisWu0 committed:
Merge branch 'develop' into feature/AC-10-create-user-signup-login-flow

NorrisWu0 committed:
Clean up code format

NorrisWu0 committed:
Refactor `signup` to `sign-up`

NorrisWu0 committed:
Update SignUpForm with refactored form components

NorrisWu0 committed:
Add login option from sign up page

NorrisWu0 committed:
Fix styling issue at forget password page

NorrisWu0 committed:
Stop removing docker image on ECR

NorrisWu0 committed:
Merge branch 'develop' into feature/AC-10-create-user-signup-login-flow
